lib/ui/painting/image_generator_registry_unittests.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/image_generator_registry.h"
+
+#include "flutter/fml/mapping.h"
+#include "flutter/shell/common/shell_test.h"
+#include "flutter/testing/testing.h"
+
+namespace flutter {
+namespace testing {
+
+static sk_sp<SkData> LoadValidImageFixture() {
+  // Load the fixture with fml mappings.
+  auto fixtures_directory =
+      fml::OpenDirectory(GetFixturesPath(), false, fml::FilePermission::kRead);
+  auto fixture_mapping = fml::FileMapping::CreateReadOnly(
+      fixtures_directory, "DashInNooglerHat.jpg");
+
+  // Remap to sk_sp<SkData>.
+  SkData::ReleaseProc on_release = [](const void* ptr, void* context) -> void {
+    delete reinterpret_cast<fml::FileMapping*>(context);
+  };
+  auto data = SkData::MakeWithProc(fixture_mapping->GetMapping(),
+                                   fixture_mapping->GetSize(), on_release,
+                                   fixture_mapping.get());
+  fixture_mapping.release();
```

Comment:
This is probably fine but it'd be better to release the unique_ptr reference on null check of `data`.

Reply:
Done.